### ISSUE TITLE
Get raw response while WS fails and "Unable to Parse Json" raises

### ISSUE THREAD

**Issue by muhammad-naderi** [koush/ion#649]
it would be nice in case the "unable to parse json" raises, the call back returns the raw response, either raw or maybe in a fake json model as the result

**Comment by WasimMemon** [comment]
in case of 500 or in some condition it is not possible to send json but with ion you can display own dialog box with proper message.
Like this

<pre>
Ion.with(this)
    .load(url)
    .progressDialog(pd)
    .setLogging(TAG, Log.ERROR)
    .setTimeout(CONNECTION_TIMEOUT)
    .onHeaders(new HeadersCallback() {
        @Override
        public void onHeaders(HeadersResponse headers) {
            if (headers.code() == HttpURLConnection.HTTP_INTERNAL_ERROR)
            {
                // display dialog with proper message
                return;
            }
        }
    })
    .setJsonObjectBody(createJsonObject(dataModel))
    .asJsonObject()
    .setCallback(new FutureCallback<JsonObject>() {
        @Override
        public void onCompleted(Exception e, JsonObject result) {
            if (pd.isShowing()) {
                pd.cancel();
            }

            if (e != null) {

                if (e instanceof TimeoutException)
                {
                    // show time out toast message
                    return;
                }

               
            } else {
                Log.d(TAG, "Service Response:-" + result.toString());
            }
        }
    });

</pre>

**Comment by ianwhite2** [comment]
@koush  @WasimMemon  does .onHeaders(new HeadersCallback() works for no 2XX response ?
For me its not getting called

**Comment by WasimMemon** [comment]
@ianwhite2 Yes it works fine for me. as you can see i have check for not 2XX response in my code. which version you are using ?

**Comment by ianwhite2** [comment]
@WasimMemon  using the latest one only . The only difference between yours and mine is code is that I am not using  .setLogging(TAG, Log.ERROR). Is that be the reason ?

**Comment by ianwhite2** [comment]
@WasimMemon @koush   can you also suggest some better way to handle Internet connection loss exception to be showed while  in middle of request ?

**Comment by WasimMemon** [comment]
@ianwhite2 no .setLogging() is for log purpose it can't affect response or headers.  Try to print headers and see what exactly you are getting in headers then do code for handle it.

**Comment by ianwhite2** [comment]
@WasimMemon ok thanks   how to handle internet connection loss  in middle of api request .When I make a request at that time I check if internet is available but I don't know how to handle when it happens when its in middle of making request and getting response Any help you can provide on this ?
